### ISSUE TITLE
Accuracy Drop in Zero-Shot Setting After Removing Distributed GPU Settings in main_nce.py

### ISSUE THREAD

**Issue by miyajiyuta** [tomchen-ctj/OST#2]
Dear @tomchen-ctj

Thank you for releasing the checkpoints!

I have been testing the inference code in a zero-shot setting and would like to perform inference without distributed GPUs. I modified the [main_nce.py](https://github.com/tomchen-ctj/OST/blob/main/main_nce.py) script to test it with the UCF-101 dataset. However, the accuracy dropped from 79.7% to 68.7%.

Here are the changes I made:

- Removed: [torch.nn.parallel.DistributedDataParallel](https://github.com/tomchen-ctj/OST/blob/1810ae6e56ebcb549b333ea349a8ca80d41a62d1/main_nce.py#L147)
- Removed: [init_distributed](https://github.com/tomchen-ctj/OST/blob/1810ae6e56ebcb549b333ea349a8ca80d41a62d1/main_nce.py#L329-L336)
- Changed: [model.module.inference](https://github.com/tomchen-ctj/OST/blob/1810ae6e56ebcb549b333ea349a8ca80d41a62d1/main_nce.py#L298) to `model.inference`
- Changed:   [logger settings](https://github.com/tomchen-ctj/OST/blob/1810ae6e56ebcb549b333ea349a8ca80d41a62d1/main_nce.py#L350C54-L350C79) to `dist_rank=-1`

Could you please help me understand what might have caused the drop in accuracy?

Thank you for your assistance!

**Comment by tomchen-ctj** [comment]
Hi @miyajiyuta, sorry for the late reply.

First, have you successfully achieved the reported accuracy in our paper using DDP?

We haven't yet tested inference without distributed GPUs, we will try to find time to dig into this problem. But still, we strongly suggest you implement OST in distributed settings first.

**Comment by miyajiyuta** [comment]
@tomchen-ctj 
thank you for reply！！


I have tested  inference with distributed settings and achieved the reported accuracy in your paper successfully！

I am looking forward to hearing further information！

thank you！
